microcontrollers: start adding new boards arduino zero, esp32, esp826… (#114)
tinygo-org/tinygo-site: content/microcontrollers/esp32.md

---
title: "ESP32"
weight: 3
---

> The [Espressif ESP32]() is a tiny development board based on the Xtensa [ESP32]() family of SoC.

Reviewer: What about this?

```suggestion
The [Espressif ESP32](https://www.espressif.com/en/products/socs/esp32) is a powerful chip that is used on many different development boards.
```

Author: Done.

## Interfaces

| Interface | Hardware Supported | TinyGo Support |
| --------- | ------------- | ----- |
| GPIO      | YES | YES |
| UART      | YES | YES |
| SPI      | YES | YES |
| I2C      | YES | Not Yet |
| ADC      | YES | Not Yet |

> | PWM      | YES | Not Yet |

Reviewer: Perhaps you could add Bluetooth and WiFi in here, just to show that they aren't supported yet.

Author: Done.

## Machine Package Docs

[Documentation for the machine package for the ESP32](../machine/esp32)

## Flashing

### CLI Flashing on Linux

You need to install the Espressif toolchain for Linux to use TinyGo with the ESP32: 

https://docs.espressif.com/projects/esp-idf/en/release-v3.0/get-started/linux-setup.html#standard-setup-of-toolchain-for-linux

In addition, you must install the `esptool` flashing tool:

https://github.com/espressif/esptool#easy-installation

Now you should be able to flash your board as follows:

- Plug your ESP32 board into your computer's USB port.
- Build and flash your TinyGo code using the `tinygo flash` command. This command flashes the ESP32 with the blinky1 example:

    ```

>     tinygo flash -target=esp32-wroom-32 examples/blinky1

Reviewer: ```suggestion
    tinygo flash -target=esp32-wroom-32 -port=/dev/ttyUSB0 examples/blinky1
```

This is only for Linux. I should fix that eventually.

Author: You mean the default port will work for other platforms?

Reviewer: It should, yes. I haven't verified this.

Author: OK, thanks.
